### PR TITLE
Issue626 store results

### PR DESCRIPTION
**Pull request by HWalnum**
@dhblum. Here is a first version of implementation of automatic storing results.
Please comment, both on the implementation and which parameters should be stored in results.json

### CODE REVIEW

File: testcase.py [hunk 1]
```diff
@@ -1327,6 +1330,44 @@ def _get_full_current_state(self):
...
+
+
+        Returns
+        -------
+        None.
+
+        '''
+
+        name = "results"
+        # get results_json
```
Comment (dhblum) [inline testcase.py:1346]: To reduce code duplication, I suggest abstracting the creation of a results.json structure in a separate function for all fields that are common with what is submitted to dashboard, defined here https://github.com/ibpsa/project1-boptest/blob/master/testcase.py#L1143.  Then, ``post_results_to_dashboard`` and ``store_results`` can share use of that function to build an initial json and add additional fields as-needed  (if any) for submission to dashboard or writing to file, respectively.

File: testcase.py [hunk 2]
```diff
@@ -1327,6 +1330,44 @@ def _get_full_current_state(self):
...
+        be retrieveable with the test_id
+
+
+        Returns
+        -------
+        None.
+
+        '''
+
+        name = "results"
```
Comment (dhblum) [inline testcase.py:1345]: Suggest calling "file_name" instead of "name"

**Review by dhblum — CHANGES_REQUESTED** [review]
Thank you @HWalnum.  Please see my inline comments and address.  Also, can you add a note to the ``releasenotes.md`` under the backwards compatible section?  And can you re-target this PR to ``ibpsa:issue626_storeResults`` instead of ``ibpsa:master``?

File: testcase.py [hunk 3]
```diff
@@ -1327,6 +1330,44 @@ def _get_full_current_state(self):
...
+        # get results_json
+        results_json = {
+            "dateRun": str(datetime.now(tz=pytz.UTC)),
+            "boptestVersion": self.version,
+            "emulatorName": self.get_name()[2]['name'],
+            "controlStep": str(self.get_step()[2]),
+            "forecastParameters": {},  # for future use to store used parameters?
+            "measurementParameters": {},  # for future use to store used parameters?
```
Comment (dhblum) [inline testcase.py:1353]: I have an initial implementation described here: https://github.com/ibpsa/project1-boptest/issues/514, but ran into an issue for truly accurate tracking as described at the end of that issue.

**Comment by dhblum** [comment]
Noting this is for https://github.com/ibpsa/project1-boptest/issues/626.

File: testcase.py (hunk 1 above)
Comment (HWalnum) [reply testcase.py:1357]: I proposed a solution. However, I had some trouble with submitting results to the dashboard, with both previous and updated solution, so I have not checked that it works properly.

File: testcase.py (hunk 3 above)
Comment (HWalnum) [reply testcase.py]: Ok. I support the last proposed solution. But I guess this could be implemented first anyway?

File: testcase.py (hunk 1 above)
Comment (dhblum) [reply testcase.py:1357]: @HWalnum I'll take a look, but looks like you got a couple results submitted for BESTEST Hydronic Heat Pump under the account "hwaln" at https://dashboard.dev.boptest.net/.

Comment (HWalnum) [reply testcase.py:1357]: Thanks @dhblum. I think those are from some extra tests I did with the service.

**Comment by dhblum** [comment]
This is looking good.  I will merge this to the staging branch and make a few adjustments there.

File: testcase.py (hunk 1 above)
Comment (dhblum) [reply testcase.py:1357]: Ok actually I'm having some trouble submitting to the dashboard too.

Comment (dhblum) [reply testcase.py:1357]: And Ok I see.  I'm meeting with Kyle tomorrow morning for a few things and he can help work out what's going on.

**Pull request merged**